plumbing/format/gitignore/dir.go

```diff
@@ -61,7 +61,9 @@ func ReadPatterns(fs billy.Filesystem, path []string) (ps []Pattern, err error)
 	for _, fi := range fis {
 		if fi.IsDir() && fi.Name() != gitDir {
 			var subps []Pattern
-			subps, err = ReadPatterns(fs, append(path, fi.Name()))
+			tempPath := make([]string, len(path))
```

Comment:
A simpler fix is `subps, err = ReadPatterns(fs, append(path[0:len(path):len(path)], fi.Name()))`